apple/Sources/FerrostarCore/Speech/SpokenInstructionObserver.swift

```diff
@@ -0,0 +1,74 @@
+import AVFoundation
+import Combine
+import FerrostarCoreFFI
+import Foundation
+
+/// An Spoken instruction provider that takes a speech synthesizer.
+public class SpokenInstructionObserver: ObservableObject {
+    @Published public private(set) var isMuted: Bool
+
+    private let synthesizer: SpeechSynthesizer
+
+    /// Create a spoken instruction observer with any ``SpeechSynthesizer``
+    ///
+    /// - Parameters:
+    ///   - synthesizer: The speech synthesizer.
+    ///   - isMuted: Whether the speech synthesizer is currently muted. Assume false if unknown.
+    public init(
+        synthesizer: SpeechSynthesizer,
+        isMuted: Bool
+    ) {
+        self.synthesizer = synthesizer
+        self.isMuted = isMuted
+    }
+
+    public func spokenInstructionTriggered(_ instruction: FerrostarCoreFFI.SpokenInstruction) {
+        guard !isMuted else {
+            return
+        }
+
+        let utterance: AVSpeechUtterance = if #available(iOS 16.0, *),
+                                              let ssml = instruction.ssml,
+                                              let ssmlUtterance = AVSpeechUtterance(ssmlRepresentation: ssml)
+        {
+            ssmlUtterance
+        } else {
+            AVSpeechUtterance(string: instruction.text)
+        }
+
+        synthesizer.speak(utterance)
+    }
+
+    /// Toggle the mute.
+    public func toggleMute() {
+        // TODO: We flip the publisher before actually stopping the synthesizer for a responsive.
+        //      But this still seems a little jumpy/slow. We may want to use some Tasks/concurrency
+        //      to separate UI on main, and speech on another queue.
+        let isCurrentlyMuted = isMuted
+        isMuted = !isCurrentlyMuted
+
+        // This used to have `synthesizer.isSpeaking`, but I think we want to run it regardless.
```

Comment:
Yeah, probably fine. I can test.

Reply:
This is quite a regression... will try to see if I can improve it before merge.

Reply:
I think I've got something "good enough" for now. Though I am honestly not quite sure why it feels slightly less responsive than other buttons.